### PR TITLE
feat: add style guideline, drop unused code

### PR DESCRIPTION
**Pull request by eordano**
This PR introduces the file https://github.com/decentraland/unity-renderer/blob/feat/style/browser-interface/STYLE.md and incorporates most of its feedback:

* drops dead code paths, reduces unused exports
* improves on multiple `yield select(...)` calls used in some sagas
* moves some `await a; await b;` calls to be `await Promise.all([a, b])`
* moves functions that are not the main export to the bottom of the files
* uses `jsonFetch` instead of `a = await fetch(); await a.json();` <-- This might be slightly controversial, as results are cached.

Test plan:
- No user-facing behavior should be detected. This erases ~1,000 lines of code from `packages/shared` that were not used, about 4% of the 20,000 LoC there!

### CODE REVIEW

File: browser-interface/packages/shared/comms/sceneSubscriptions.ts [hunk 1]
```diff
@@ -14,18 +14,3 @@ export function subscribeParcelSceneToCommsMessages(controller: ICommunicationsC
 export function unsubscribeParcelSceneToCommsMessages(controller: ICommunicationsController) {
   scenesSubscribedToCommsEvents.delete(controller)
 }
-
-/**
- * Retrieve the scene IDs that are subscribed to receive scene messages from comms
- *
- * @returns a list of CIDs, identifying the Scenes by their hash
- */
-export function getParcelSceneSubscriptions(): string[] {
```
Comment (menduz) [inline browser-interface/packages/shared/comms/sceneSubscriptions.ts:23]: Not used? This is supposed to be used to make the MessageBus work

**Review by menduz — APPROVED** [review]
Awesome job! I'm hesitant about `{a} = yield select` because of the lack of types for `a`

File: browser-interface/packages/shared/comms/sceneSubscriptions.ts (hunk 1 above)
Comment (eordano) [reply browser-interface/packages/shared/comms/sceneSubscriptions.ts:23]: https://sourcegraph.com/search?q=context%3A%40eordano%2Fdecentraland+getParcelSceneSubscriptions&patternType=standard&sm=1&groupBy=repo

**Comment by eordano** [comment]
> Awesome job! I'm hesitant about `{a} = yield select` because of the lack of types for `a`. Did you think about any solution?

You're right! Adding comments about being explicit on the return type. Check out https://github.com/decentraland/unity-renderer/pull/4498/commits/1da6a247ed0d4adbee812acc28e1aab7f801376e

**Review by kwssbocian — APPROVED** [review]
✔️No changes observed from the user side.

File: browser-interface/packages/shared/friends/sagas.ts [hunk 2]
```diff
@@ -551,7 +551,7 @@ function* initializePrivateMessaging() {
 
   if (profile) {
     const displayName = calculateDisplayName(profile)
-    yield apply(client, client.setProfileInfo, [{ displayName }])
+    yield call(async () => await client.setProfileInfo({ displayName }))
```
Comment (eordano) [inline browser-interface/packages/shared/friends/sagas.ts:554]:
```suggestion
    yield apply(client, client.setProfileInfo, [{ displayName }])
```

File: browser-interface/packages/shared/friends/sagas.ts [hunk 3]
```diff
@@ -640,7 +640,7 @@ function* refreshFriends() {
       .filter((each, i, elements) => elements.indexOf(each) === i)
 
     const ensureFriendProfilesPromises = allProfilesToObtain.map((userId) => ensureFriendProfile(userId))
-    yield Promise.all(ensureFriendProfilesPromises).catch(logger.error)
+    yield call(async () => await Promise.all(ensureFriendProfilesPromises).catch(logger.error))
```
Comment (eordano) [inline browser-interface/packages/shared/friends/sagas.ts:643]:
```suggestion
    yield call(ensureFriendProfilesPromises)
```

File: browser-interface/packages/shared/friends/sagas.ts [hunk 4]
```diff
@@ -578,7 +578,7 @@ function* refreshFriends() {
     const friendsSocial: SocialData[] = []
 
     // init friend requests
-    const friendRequests: FriendshipRequest[] = yield client.getPendingRequests()
+    const friendRequests: FriendshipRequest[] = yield call(async () => await client.getPendingRequests())
```
Comment (eordano) [inline browser-interface/packages/shared/friends/sagas.ts:581]:
```suggestion
    const friendRequests: FriendshipRequest[] = yield apply(client, client.getPendingRequests)
```

**Pull request closed without merging**